TESTING.md

```diff
@@ -0,0 +1,104 @@
+Jetty Testing Design
+==
+
+The goal of testing in the context of the vanilla runtime is to provide an extensible framework with two modes, one for local testing and validation of an application and one for remote testing which deploys applications as services into the GCE.  Tests are not a part of the normal build process, they are a separate process with both local and remote components.  Additionally tests are conceptually disconnected from the actual build of the jetty-runtime project and should be considered as something that can pulled into their own github repository with minimal changes required to the core jetty-runtime build or project.
+
+
+Execution Examples:
+===
+
+Below are a series of commands and their expected behaviors.
+
+
+Standard Execution
+====
+
+```
+> mvn install
+```
+
+
+This should run the normal build cycle for the jetty-runtime project.  No tests are expected to run during this step.  The docker image being built should be installed locally minimally to the tag jetty:9.x.
+
+```
+> mvn deploy
```

Comment:
I don't think this works to deploy to GCR. It would only attempt to deploy to Maven Central, or am I missing something?
The way I deploy to GCR is:
```
$ docker tag jetty:9.3 gcr.io/{project}/jetty:9:4
$ gcloud docker push gcr.io/{project}/jetty:9:4 
```